src/array.rs

```diff
@@ -306,6 +305,85 @@ impl<T: TreeLike + fmt::Display> fmt::Display for BinaryTree<T> {
     }
 }
 
+/// One step on a path from ancestor nodes to descendent nodes.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum Direction {
```

Comment:
In d1d5c78388208d4e16a60a2059044b42c01ba7e0:

I believe this enum should not be `pub` because it's an implementation detail of `DirectedTree`.

Reply:
Ahh, no, it is returned from `find`. Interesting.

I wonder if you want to encapsulate the returned `Vec<Direction>` into a `DirectedTreePath` or something which hides the `Direction` type. Would result in a simpler API and be easier to reason about where direction-vectors come from.

Sadly in Rust you can't tie the `DirectedTreePath` to a specific directed tree the way you could with a dependent type system. (You could put a reference to the `DirectedTree` into the path but that's super unergonomic because it prevents moving the tree. And you could do some crazy run-time "tree ID" thing, maybe gated to only compile in debug mode, but that's a lot of work and shouldn't be in this PR.)